src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabMergeRequestSCMEvent.java

```diff
@@ -119,41 +115,46 @@ public String description() {
             MergeRequestEvent.ObjectAttributes m = getPayload().getObjectAttributes();
             String originOwner = getPayload().getUser().getUsername();
             String originProjectPath = m.getSource().getPathWithNamespace();
-            Set<ChangeRequestCheckoutStrategy> strategies = request.getMRStrategies(
-                    source.getProjectPath().equalsIgnoreCase(originProjectPath)
-            );
-            for (ChangeRequestCheckoutStrategy strategy : strategies) {
-                MergeRequestSCMHead h = new MergeRequestSCMHead(
-                        "MR-" + m.getIid() + (strategies.size() > 1 ? "-" + strategy.name()
-                                .toLowerCase(Locale.ENGLISH) : ""),
-                        m.getId(),
-                        new BranchSCMHead(m.getTargetBranch()),
-                        ChangeRequestCheckoutStrategy.MERGE,
-                        originProjectPath.equalsIgnoreCase(source.getProjectPath())
-                                ? SCMHeadOrigin.DEFAULT
-                                : new SCMHeadOrigin.Fork(originProjectPath),
-                        originOwner,
-                        originProjectPath,
-                        m.getSourceBranch());
-                GitLabApi gitLabApi = GitLabHelper.apiBuilder(source.getServerName());
-                // Since rn there is no way to get the commit hash of target branch fetch the mr again
-                // A better way can be searching for the target branch head and getting the commit hash
-                MergeRequest mr = gitLabApi.getMergeRequestApi().getMergeRequest(source.getProjectPath(), m.getIid());
+            int count = originProjectPath.length();
+            for(int i = originProjectPath.length()-1; i >= 0; i--) {
+                count--;
+                if(originProjectPath.charAt(i) == '/') {
```

Comment:
What? :confused:

Reply:
originProjectPath.indexOf('/') does the same thing?

Reply:
Sorry about that poor code, actually we need just the path and not the project name. So this method removes the project name from it. But if I do `indexOf` it will perhaps return the first time it encounters `\`. I think `lastIndexOf` will work.